tests/integration/service_test.go

```diff
@@ -101,4 +102,39 @@ var _ = Describe("odoServiceE2e", func() {
 	// 	})
 	// })
 
+	Context("When working from outside a component dir", func() {
+		JustBeforeEach(func() {
+			project = helper.CreateRandProject()
+			context = helper.CreateNewContext()
+			app = helper.RandString(7)
+			originalDir = helper.Getwd()
+			helper.Chdir(context)
+		})
+		JustAfterEach(func() {
+			helper.DeleteProject(project)
+			helper.DeleteDir(context)
+			helper.Chdir(originalDir)
+		})
+		It("should be able to list services in a given app and project combination", func() {
+			// create a component by copying the example
+			helper.CopyExample(filepath.Join("source", "nodejs"), context)
+			helper.CmdShouldPass("odo", "create", "nodejs", "--app", app, "--project", project)
+
+			// create a service from within a component directory
+			helper.CmdShouldPass("odo", "service", "create", "dh-postgresql-apb", "--plan", "dev",
+				"-p", "postgresql_user=luke", "-p", "postgresql_password=secret",
+				"-p", "postgresql_database=my_data", "-p", "postgresql_version=9.6",
+				"--app", app, "--project", project,
+			)
+			ocArgs := []string{"get", "serviceinstance", "-o", "name"}
+			helper.WaitForCmdOut("oc", ocArgs, 1, true, func(output string) bool {
+				return strings.Contains(output, "dh-postgresql-apb")
+			})
+
```

Comment:
Add the steps for listing out the service in the component dir to make sure it works form component dir too.